File: 백준/Silver I/6588. 골드바흐의 추측/골드바흐의 추측.cc
#include <iostream>
 
using namespace std;
 
const int MAX = 1000000;
 
bool primeArr[MAX] = {false, };
 
void primeChk(){
    
    for(int i = 2; i*i<=MAX; i++){
        
        if(!primeArr[i]){
        
            for(int j = i*i; j<=MAX; j+=i){
                primeArr[j] = true;
            }
        }
        
    }
}
 
int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    
    int n;
    primeChk();
 
    
    while(1){
        
        cin >> n;
        if(n==0){
            break;
        }
                   
        bool chk = false;
        
        for(int a = 3; a<= n; a+=2){
            if(!primeArr[a] && !primeArr[n-a]) {
                cout << n << " = " << a << " + " << n-a << "\n";
                chk = true;
                break;
            }
        }
        if(!chk){
            cout << "Goldbach's conjecture is wrong.\n";
        }
    }
  
    
    return 0;
}

File: 백준/Silver I/6588. 골드바흐의 추측/골드바흐의 추측.py
from sys import stdin
input = stdin.readline

sosu = [True for i in range(1000001)]

for i in range(2, int(1000001 ** 0.5) + 1):
    if not sosu[i]:
        continue

    for j in range(i + i, 1000001, i):
        sosu[j] = False

while True:
    buff = int(input())

    if buff == 0:
        break

    for i in range(3, buff // 2 + 1, 2):
        if sosu[i] and sosu[buff - i]:
            print(f"{buff} = {i} + {buff - i}")
            break
    else:
        print("Goldbach's conjecture is wrong.")